target/linux/ramips/image/mt7620.mk

```diff
@@ -488,6 +488,25 @@ define Device/hnet_c108
 endef
 TARGET_DEVICES += hnet_c108
 
+define Device/hootoo_tm05
+  SOC := mt7620n
+  IMAGE_SIZE := 6144k
+  DEVICE_VENDOR := HooToo
+  DEVICE_MODEL := TM05
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-i2c-ralink
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0xFD200000
+  COMPILE := loader-$(1).bin
+  COMPILE/loader-$(1).bin := loader-okli-compile | pad-to 64k | lzma | \
+	uImage lzma
+  KERNEL := $(KERNEL_DTB) | uImage lzma -M 0x4f4b4c49
+  KERNEL_INITRAMFS := $(KERNEL_DTB) | uImage lzma
+  IMAGES += kernel.bin rootfs.bin
+  IMAGE/kernel.bin := append-loader-okli $(1)
+  IMAGE/rootfs.bin := $$(sysupgrade_bin) | check-size
```

Comment:
Instead of this being = to sysupgrade.bin, it should be the same recipe but without `append-metadata`

Reply:
> it should be the same recipe but without `append-metadata`

Sorry I just don't get it!

The `IMAGE/rootfs.bin` above is doing the exact same thing what you wrote:
* it starts with `$$(sysupgrade_bin)`, defined in `Device/Default` in [`Makefile#L175`](https://github.com/openwrt/openwrt/blob/d5a148f5c8f6d6dee7041fd348bd0f52839d0f4e/target/linux/ramips/image/Makefile#L175)
* which in turn expands to `append-kernel | append-rootfs | pad-rootfs`
* it appends `check-size` next
* and we get: `append-kernel | append-rootfs | pad-rootfs | check-size`
* which is `  IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | append-metadata | check-size` recipe without `append-metadata`

Reply:
It worth noting that this `sysupgrade_bin` recipe was introduced by commit 478f1f6b167c6b1c91f5915e9567814350865c4c.

Reply:
Oops I didn't see that. Nothing like that in ath79 (yet)

Reply:
> Nothing like that in ath79 (yet)

As it is a "compatibility hack" :disappointed: 
See the introducing commit above!

Reply:
BTW nobody forces you to use these short expressions like sysupgrade_bin. Just write the commands you want to use and you are done.

Reply:
So all OK here? Thanks!